web/app/tournament/tourny.py

```diff
@@ -169,3 +172,196 @@ def select_player(tournament_id, tournament_player_id):
+    matches = Match.query.filter_by(tournament_id=tournament_id,
+                                    t_round=round_id)
+    if any_matches > 0:
+        return render_template('tournament_matches_by_round.html',
+        tournament_id=tournament_id,
+        matches=matches,
+        round_id=round_id,
+        latest_round=latest_round)
+    else:
+        #  flash('URL Error')
+        return redirect(url_for('.show', tournament_id=tournament_id))
+
+
+@tournament.route('/<tournament_id>/full_pairing/', methods=['POST'])
+def full_pairing(tournament_id):
+    """Currently the full_pairing method takes all tournament players,
+    shuffles them, and pairs them each into a match object"""
+    tournament = Tournament.query.get(tournament_id)
+    players = TournamentPlayer.query.filter_by(tournament_id=tournament_id)
+    players_list = list(players)
+    shuffle(players_list)
+
+    tourny_round = Round(tournament_id=tournament_id)
+    db.session.add(tourny_round)
+    db.session.commit()
+    tourny_round = Round.query.all()[-1]
+    while len(players_list) > 0:
+        if len(players_list) % 2 == 0:
+            match = Match(player_1_aga_num = players_list.pop().aga_num,
+                          player_2_aga_num = players_list.pop().aga_num,
+                          tournament_id = tournament_id,
+                          pairing = tournament.pairing,
+                          t_round = tourny_round.id)
+            db.session.add(match)
+            db.session.commit()
+            match = Match.query.all()[-1]
+            match.player_1_name = players.filter_by(
+                aga_num=match.player_1_aga_num).first().name
+            match.player_2_name = players.filter_by(
+                aga_num=match.player_2_aga_num).first().name
+            db.session.add(match)
+            db.session.commit()
+        else:
+            match = Match(player_1_aga_num = players_list.pop().aga_num,
+                        #   player_2_aga_num = "",
+                          tournament_id = tournament_id,
+                          pairing = tournament.pairing,
+                          t_round = tourny_round.id)
+            db.session.add(match)
+            db.session.commit()
+            match = Match.query.all()[-1]
+            match.player_1_name = players.filter_by(
+                aga_num=match.player_1_aga_num).first().name
+            # match.player_2_name = "BYE"
+            db.session.add(match)
+            db.session.commit()
+    return redirect(url_for('.matches_by_round_index',
+    tournament_id=tournament_id,
+    round_id=tourny_round.id))
+
+
+#  treating empty string returned by form as equal to None in DB
+def empty_player(none, empty_string):
+    if (none == None and empty_string == '') or (none == '' and empty_string == None):
+        return True
+    else:
+        return False
+
+
+@tournament.route('/<tournament_id>/match/<match_id>/', methods=['POST'])
+def match(tournament_id, match_id):
```

Comment:
Can you move the match logic out of the HTTP handling into a separate file? In other words, you'd still have match(...) as an HTTP method, but it would defer all logic to another method.

Typically, I think it's better style to have these HTTP handlers be light, deferring logic to elsewhere. In some sense, we've squashed all our MVC logic into one method.